packages/plugin-pack/sources/index.ts

```diff
@@ -10,9 +11,17 @@ export interface Hooks {
     project: Project,
     definePath: (path: string | null) => void,
   ) => Promise<void>,
+
+  beforeWorkspacePacking?: (
```

Comment:
In https://github.com/yarnpkg/berry/issues/137#issuecomment-490647425 you mentioned putting this hook or at least its implementation in `plugin-essentials`.
Imo it makes more sense to put the hook definition here, as the `packUtils` actually call the hook. I placed the main/module hook here as well, do you want me to move it to plugin-essentials instead?